common/app/views/fragments/preRenderJavaScript.scala.html

```diff
@@ -0,0 +1,58 @@
+@()
+
+@* This script must come after the pre-flight shims *@
+
+<script>
+(function applyAdFree(){
+    /**
+     * Determine if user has 'adfree' view, and apply appropriate body classes before rendering.
+     */
+
+    var ADFREE_COOKIE_KEY = 'gu_adfree_user',
+        USER_SIGNIN_COOKIE = 'GU_U',
+        commercial = guardian.config.commercial;
+
+    if (showAdfreeView()) {
+        addClass(document.documentElement, 'adfree');
+        commercial.showingAdfree = true;
+    } else {
+        commercial.showingAdfree = false;
+    }
+
+    function showAdfreeView() {
+        if (featureDisabled() || userLoggedOut()) {
+            return false;
+        } else {
+            var adfreeUser = readCookie(ADFREE_COOKIE_KEY);
+            // If the cookie is ambiguous or null, err on the side of caution and hold off showing ads
+            return adfreeUser !== 'false';
+        }
+    }
+
+    function featureDisabled() {
+        return guardian.config.switches.advertOptOut === false;
```

Comment:
could the whole thing be wrapped behind this check? i.e. duck out of this ASAP? same could apply for existence of the cookie, and then finally the value of the cookie?